metricbeat/magefile.go

```diff
@@ -186,3 +202,28 @@ func PythonIntegTest(ctx context.Context) error {
 		return devtools.PythonNoseTest(devtools.DefaultPythonTestIntegrationArgs())
 	}, devtools.ListMatchingEnvVars("NOSE_")...)
 }
+
+// CreateMetricset creates a new metricset.
+//
+// Required ENV variables:
+// * MODULE: Name of the module
+// * METRICSET: Name of the metricset
+func CreateMetricset(ctx context.Context) error {
+	ve, err := devtools.PythonVirtualenv()
+	if err != nil {
+		return err
+	}
+	path, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	pythonEnv := map[string]string{
+		"VIRTUAL_ENV": ve,
+	}
+	_, err = sh.Exec(
+		pythonEnv, os.Stdout, os.Stderr, "python3", "scripts/create_metricset.py",
```

Comment:
Use 

`python, err := devtools.LookVirtualenvPath(ve, "python")`

to get the python executable. It handles some weird issues (mostly windows).

I don't think you need to worry about the `VIRTUAL_ENV` var.

Reply:
Updated.